ts/features/wallet/onboarding/bancomat/screens/hooks/useImageResize.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from "react";
+import { Image } from "react-native";
+import { fromNullable } from "fp-ts/lib/Option";
+
+export type ImageDimensions = {
+  w: number;
+  h: number;
+};
+
+/**
+ * To keep the image bounded in the predefined maximum dimensions (40x160) we use the resizeMode "contain"
+ * and always calculate the resize width keeping fixed the height to 40, in this way all images will have an height of 40
+ * and a variable width until the limit of 160.
+ * Calculating the new image height based on its width may cause an over boundary dimension in some case.
+ * @param width
+ * @param height
+ * @param maxW
+ * @param maxH
+ * @param callback
+ */
+const handleImageDimensionSuccess = (
+  width: number,
+  height: number,
+  maxW: number,
+  maxH: number,
+  callback: (dim: ImageDimensions) => void
+) => {
+  if (width > 0 && height > 0) {
+    const ratio = Math.min(maxW / width, maxH / height);
+    callback({ w: width * ratio, h: height * ratio });
+  }
+};
+
+export const useImageResize = (
```

Comment:
Instead of a callback we can use an internal state and return it.
Could this code be an improvement?

```typescript
const handleImageDimensionSuccess = (
  width: number,
  height: number,
  maxW: number,
  maxH: number
): [number, number] => {
  if (width > 0 && height > 0) {
    const ratio = Math.min(maxW / width, maxH / height);
    return [width * ratio, height * ratio];
  }
  return [width, height];
};

export const useImageResize = (
  maxWidth: number,
  maxHeight: number,
  imageUrl?: string
): [number, number] => {
  const [size, setSize] = useState<[number, number]>([0, 0]);
  useEffect(() => {
    fromNullable(imageUrl).map(url =>
      Image.getSize(url, (w, h) =>
        setSize(handleImageDimensionSuccess(w, h, maxWidth, maxHeight))
      )
    );
  }, [imageUrl]);
  return size;
};
```

usage
```typescript
const imageDimensions = useImageResize(
    BASE_IMG_W,
    BASE_IMG_H,
    props.bank.logoUrl
  );

  const imageStyle: StyleProp<ImageStyle> = {
    width: imageDimensions[0],
    height: imageDimensions[1],
    resizeMode: "contain"
  };
```

Reply:
PS I didn't test it

Reply:
Yes, maybe an external (from the component point of view) state could be useful, I'll give it a try

Reply:
@Undermaken works delightfully, implemented in d6a39b2